easybuild/easyblocks/c/comsol.py

```diff
@@ -0,0 +1,136 @@
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Comsol(PackedBinary):
+    """Support for installing COMSOL."""
+
+    def __init__(self, *args, **kwargs):
+        """Add extra config options specific to COMSOL."""
+        super(EB_Comsol, self).__init__(*args, **kwargs)
+        self.comp_fam = None
+        self.configfile = os.path.join(self.builddir, 'my_setupconfig.ini')
+
+    def configure_step(self):
+        """Configure COMSOL installation: create license file."""
+
+        # The tar file comes from the DVD and has 0444 as permission at
+        # the top dir.
+        adjust_permissions(self.cfg['start_dir'], stat.S_IWUSR)
+
+        default_lic_env_var = 'LM_LICENSE_FILE'
+        lic_specs, self.license_env_var = find_flexlm_license(custom_env_vars=[default_lic_env_var],
+            lic_specs=[self.cfg['license_file']])
+
+        if lic_specs:
+            if self.license_env_var is None:
+                self.log.info("Using Comsol license specifications from 'license_file': %s", lic_specs)
+                self.license_env_var = default_lic_env_var
+            else:
+                self.log.info("Using Comsol license specifications from $%s: %s", self.license_env_var, lic_specs)
+
+            self.license_file = os.pathsep.join(lic_specs)
+            env.setvar(self.license_env_var, self.license_file)
+        else:
+            msg = "No viable license specifications found; "
+            msg += "specify 'license_file', or define $LM_LICENSE_FILE"
+            raise EasyBuildError(msg)
+
+        try:
+            shutil.copyfile(os.path.join(self.cfg['start_dir'], 'setupconfig.ini'), self.configfile)
+            config = read_file(self.configfile)
+
+            regdest = re.compile(r"^installdir\s*=.*", re.M)
+            reggui = re.compile(r"^showgui\s*=.*", re.M)
+            regagree = re.compile(r"^agree\s*=.*", re.M)
+            reglicpath = re.compile(r"^license\s*=.*", re.M)
+            reglinuxlauncher = re.compile(r"^linuxlauncher\s*=.*", re.M)
+            regsymlinks = re.compile(r"^symlinks\s*=.*", re.M)
+            regfileassoc = re.compile(r"^fileassoc\s*=.*", re.M)
+
+            config = regdest.sub("installdir=%s" % self.installdir, config)
+            config = reggui.sub("showgui=0", config)
+            config = regagree.sub("agree=1", config)
+            config = reglicpath.sub("license=%s" % self.license_file, config)
+            config = reglinuxlauncher.sub("linuxlauncher=0", config)
+            config = regsymlinks.sub("symlinks=0", config)
+            config = regfileassoc.sub("fileassoc=0", config)
+
+            matlab_root = get_software_root("MATLAB")
+            if matlab_root:
+                regmatlab = re.compile(r"^matlabdir\s*=.*", re.M)
+                config = regmatlab.sub("matlabdir=%s" % matlabroot, config)
+
+            write_file(self.configfile, config)
+
+        except IOError, err:
```

Comment:
Any `IOError` that occur are handled by `read_file`, `write_file` or `copy_file` already, so no need to wrap this in a `try` `except`.